signer/src/stacks/api.rs

```diff
@@ -320,16 +321,87 @@ impl TenureBlocks {
     }
+}
+
+/// A slimmed down [`NakamotoBlockHeader`]
+#[derive(Debug, Clone, PartialEq)]
+pub struct StacksBlockHeader {
+    /// The total number of StacksBlocks and NakamotoBlocks preceding this
+    /// block in this block's history.
+    pub block_height: u64,
+    /// The identifier for a block. It is the hash of the this block's
+    /// header hash and the block's consensus hash.
+    pub block_id: StacksBlockId,
+    /// The index block hash of the immediate parent of this block. This is
+    /// the hash of the parent block's hash and consensus hash.
+    pub parent_block_id: StacksBlockId,
+}
+
+impl From<NakamotoBlockHeader> for StacksBlockHeader {
+    fn from(value: NakamotoBlockHeader) -> Self {
+        StacksBlockHeader {
+            block_height: value.chain_length,
+            block_id: value.block_id(),
+            parent_block_id: value.parent_block_id,
+        }
+    }
+}
+
+impl From<TenureBlocks> for TenureHeaders {
+    fn from(value: TenureBlocks) -> Self {
+        let headers = value
+            .blocks
+            .into_iter()
+            .map(|block| StacksBlockHeader::from(block.header))
+            .collect();
+
+        TenureHeaders {
+            headers,
+            anchor_block_hash: value.anchor_block_hash,
+            anchor_block_height: value.anchor_block_height,
+        }
+    }
+}
+
+/// This struct represents a non-empty subset of the Stacks block headers
+/// that were created during a tenure.
+#[derive(Debug)]
+pub struct TenureHeaders {
+    /// The subset of Stacks block headers that of Nakamoto blocks that
+    /// were created during a tenure. This is always non-empty.
+    headers: Vec<StacksBlockHeader>,
+    /// The bitcoin block that this tenure builds off of.
+    pub anchor_block_hash: BitcoinBlockHash,
+    /// The height of the bitcoin block associated with the above block
+    /// hash.
+    pub anchor_block_height: u64,
+}
+
+impl TenureHeaders {
+    /// Get all the headers contained in this object.
+    ///
+    /// # Note
+    ///
+    /// * The returned slice is nonempty.
+    /// * The struct doesn't need to contain all the headers of stacks
+    ///   blocks in a tenure.
+    pub fn headers(&self) -> &[StacksBlockHeader] {
+        &self.headers
+    }
 
     /// Return an iterator of Stacks blocks included in this object.
-    pub fn as_stacks_blocks(&self) -> impl Iterator<Item = StacksBlock> + '_ {
-        let bitcoin_anchor = &self.anchor_block_hash;
-        self.blocks
-            .iter()
-            .map(|block| StacksBlock::from_nakamoto_block(block, bitcoin_anchor))
+    pub fn as_stacks_blocks(self) -> impl Iterator<Item = StacksBlock> {
+        let bitcoin_anchor = self.anchor_block_hash;
+        self.headers.into_iter().map(move |header| StacksBlock {
+            block_hash: header.block_id.into(),
+            block_height: header.block_height,
+            parent_hash: header.parent_block_id.into(),
+            bitcoin_anchor,
+        })
     }
```

Comment:
I'd probably rename this to `into_iter()` or `into_xx_iter()` etc. while you're at it as it's more conventional for a method that consumes `self` and returns an `Iterator`.

Reply:
https://github.com/stacks-sbtc/sbtc/pull/1604/commits/4986f8c64bc81e8568ded96e3451fad1e71d50c8